syntax = "proto3";

package com.webank.ai.fate.api.serving;
option java_outer_classname = "PredictionServiceProto";

message FederatedMeta{
    string sceneId = 1;
    string myPartyId = 2;
    string partnerPartyId = 3;
    string myRole = 4;
    string commitId = 5;
}

message DataMap{
    map<string, float> floatData = 1;
    map<string, string> stringData = 2;
}

message PredictRequest{
    FederatedMeta meta = 1;
    map<string, DataMap> data = 5;
}

message PredictResponse{
    int32 statusCode = 1;
    FederatedMeta meta = 2;
    map<string, DataMap> data = 3;
    string dataJson = 6;
}

message FederatedPredictRequest{
    FederatedMeta meta = 1;
    string vectorId = 5;
}

message FederatedPredictResponse{
    int32 statusCode = 1;
    FederatedMeta meta = 2;
    map<string, float> result = 5;
}

service PredictionService{
    rpc predict(PredictRequest) returns (PredictResponse);
    rpc federatedPredict(FederatedPredictRequest) returns (FederatedPredictResponse);
}
